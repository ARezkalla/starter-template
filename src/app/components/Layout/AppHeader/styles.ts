const DEFAULT_SPACING_FACTOR = 8

export const styles = {
  toolbar: {
    backgroundColor: "header.main",
    display: "flex",
    justifyContent: "space-between",
    alignItems: "center",
    height: 7 * DEFAULT_SPACING_FACTOR,
    padding: "0px 16px !important",
  },
  container: {
    display: "flex",
    mr: "60px",
    justifyContent: "flex-start",
    alignItems: "center",
  },
  icon: {
    width: 6 * DEFAULT_SPACING_FACTOR,
    cursor: "pointer",
  },
  logo: {
    width: 4 * DEFAULT_SPACING_FACTOR,
    height: 2 * DEFAULT_SPACING_FACTOR,
    cursor: "pointer",
    marginRight: 8,
    marginTop: 2,
  },
  typography: {
    color: "text.primary",
    fontWeight: "700",
  },
  settings: {
    marginLeft: 3,
    padding: "0.188rem",
    backgroundColor: "header.main",
    borderRadius: "50%",
    height: "1.875rem",
    width: "1.875rem",
    border: "2px solid",
    color: "primary.main",
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    cursor: "pointer",
  },
  personIcon: {
    height: "1,5rem",
    color: "primary",
  },
}
