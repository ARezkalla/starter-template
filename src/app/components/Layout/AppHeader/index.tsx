import { Avatar, Box, Toolbar } from "@mui/material"
import PersonOutlineIcon from "@mui/icons-material/PersonOutline"

import { setIsSettingsOpen } from "../../../store/rootSlice"
import { styles } from "./styles"
import { useNavigate } from "react-router-dom"
import { useAppDispatch, useAppSelector } from "@/app/store/hooks"
import HeaderSearchBar from "./components/SearchBar"
import logoIC from "@assets/coreic.png"

const AppHeader = () => {
  const { isSettingsOpen } = useAppSelector((state) => state.global)
  const dispatch = useAppDispatch()
  const navigate = useNavigate()

  const handleSettingsOpen = () => {
    dispatch(setIsSettingsOpen(!isSettingsOpen))
  }

  const logo = logoIC

  return (
    <Toolbar sx={styles.toolbar}>
      <Box sx={styles.container}>
        <Box
          sx={styles.icon}
          component="img"
          onClick={() => navigate("dashboard")}
          src={logo}
          alt="corebeat icon"
        />
      </Box>

      <HeaderSearchBar />
      <Avatar sx={styles.settings} onClick={handleSettingsOpen}>
        <Box sx={styles.personIcon}>
          <PersonOutlineIcon />
        </Box>
      </Avatar>
    </Toolbar>
  )
}

export default AppHeader
