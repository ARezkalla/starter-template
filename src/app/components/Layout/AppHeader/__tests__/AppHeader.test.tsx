import { renderWithProviders } from "@/app/test/renderWithProvider"
import { describe, expect, it } from "vitest"
import AppHeader from ".."

describe("AppHeader", () => {
  it("should render without crashing", () => {
    const { container } = renderWithProviders(<AppHeader />)
    expect(container).toMatchSnapshot()
  })
})
