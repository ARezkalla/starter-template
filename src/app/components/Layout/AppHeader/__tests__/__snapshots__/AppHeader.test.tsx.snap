// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppHeader > should render without crashing 1`] = `
<div>
  <div
    class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1sixtck-MuiToolbar-root"
  >
    <div
      class="MuiBox-root css-1p1c9rl"
    >
      <img
        alt="corebeat icon"
        class="MuiBox-root css-sfvpng"
        src="/src/app/assets/coreic.png"
      />
    </div>
    <div
      class="MuiBox-root css-1wal5io"
    >
      <div
        class="MuiBox-root css-3g3bdw"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-18m7kj6-MuiSvgIcon-root"
          data-testid="SearchIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </div>
      <div
        class="MuiInputBase-root MuiInputBase-colorPrimary css-h0byw8-MuiInputBase-root"
      >
        <input
          aria-label="search"
          class="MuiInputBase-input css-1u71ta7-MuiInputBase-input"
          placeholder="Search"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1ifj1il-MuiAvatar-root"
    >
      <div
        class="MuiBox-root css-12f00zy"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="PersonOutlineIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 5.9c1.16 0 2.1.94 2.1 2.1s-.94 2.1-2.1 2.1S9.9 9.16 9.9 8s.94-2.1 2.1-2.1m0 9c2.97 0 6.1 1.46 6.1 2.1v1.1H5.9V17c0-.64 3.13-2.1 6.1-2.1M12 4C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 9c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
